### PR TITLE
chore: Replace routeMixin with useRouteHelper composable [CW-3476]

### PR DESCRIPTION
**Pull request by fayazara**
# Pull Request Template

## Description

Replaces the routerMixin with the new useRouterHelper composable

Fixes https://linear.app/chatwoot/issue/CW-3476/remove-routermixin-and-use-it-as-it-is-wherever-required

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/widget/App.vue [hunk 1]
```diff
@@ -8,7 +8,7 @@ import availabilityMixin from 'widget/mixins/availability';
 import { getLocale } from './helpers/urlParamsHelper';
 import { isEmptyObject } from 'widget/helpers/utils';
 import Spinner from 'shared/components/Spinner.vue';
-import routerMixin from './mixins/routerMixin';
+import { useRouterHelper } from 'widget/composables/useRouterHelper';
```
Comment (scmmishra) [inline app/javascript/widget/App.vue:11]: We can just make a composable 'useReplaceRoute' instead of `useRouterHelper` that exposes the entire fn

**Review by scmmishra — CHANGES_REQUESTED** [review]

**Comment by fayazara** [comment]
@scmmishra PTAL, refactored `useRouteHelper` to `useReplaceRoute`

File: app/javascript/widget/App.vue (hunk 1 above)
Comment (fayazara) [reply app/javascript/widget/App.vue]: Resolved in https://github.com/chatwoot/chatwoot/pull/9941/commits/a0153daca413e55b9bfd86483624cac82e8b9eaf

**Review by scmmishra — COMMENTED** [review]
Hey @fayazara sorry I wasn't very clear in the previous comment, I was suggestion having the `useReplaceRoute` do the action directly

```js
import { useRouter, useRoute } from 'dashboard/composables/route';

/**
 * Composable for replacing the current route with a new one if it's different.
 * @param {string} name - The name of the route to replace with.
 * @param {Object} params - The params to pass to the new route.
 * @returns {Promise|undefined} A promise that resolves when the navigation is complete, or undefined if no navigation occurs.
 */
export const useReplaceRoute = (name, params = {}) => {
  const router = useRouter();
  const route = useRoute();

  if (route.name !== name) {
    return router.replace({ name, params });
  }
  return undefined;
};

export default useReplaceRoute;
```

WDYT about this? If this feels a bit odd, let's keep the original implementation before a0153daca413e55b9bfd86483624cac82e8b9eaf

**Comment by fayazara** [comment]
@scmmishra reverted the commit back to useRouterHelper

File: app/javascript/widget/mixins/specs/useRouteReplace.spec.js
Comment (iamsivin) [inline app/javascript/widget/mixins/specs/useRouteReplace.spec.js:1]: @fayazara Why do we need two spec files?

File: app/javascript/widget/mixins/specs/useRouterHelper.spec.js
Comment (iamsivin) [inline app/javascript/widget/mixins/specs/useRouterHelper.spec.js:1]: @fayazara Let us move the spec folder from mixin path to composable.

**Comment by scmmishra** [comment]
@fayazara I was suggesting something like this: https://github.com/chatwoot/chatwoot/compare/chore-replace-router-mixin...refactor-replace-router-composable

PTAL and see if it sounds like a good idea?

**Comment by scmmishra** [comment]
@fayazara PTAL

File: app/javascript/widget/mixins/specs/useRouterHelper.spec.js
Comment (fayazara) [reply app/javascript/widget/mixins/specs/useRouterHelper.spec.js:1]: Fixed this and deleted the unused old spec

**Comment by scmmishra** [comment]
> @scmmishra A bit confused, do you mean just making one parent method resolving the route?
> 
> ```js
> import { useRouter, useRoute } from 'dashboard/composables/route';
> 
> /**
>  * Composable for replacing the current route with a new one if it's different.
>  * @param {string} name - The name of the route to replace with.
>  * @param {Object} params - The params to pass to the new route.
>  * @returns {Function} A function that replaces the route when called.
>  */
> export const useReplaceRoute = () => {
>   const router = useRouter();
>   const route = useRoute();
> 
>   return (name, params = {}) => {
>     if (route.name !== name) {
>       return router.replace({ name, params });
>     }
>     return undefined;
>   };
> };
> 
> export default useReplaceRoute;
> ```

Yep, since the composable doesn't do anything else

**Comment by fayazara** [comment]
@scmmishra I have updated this

File: app/javascript/widget/components/PreChat/Form.vue [hunk 2]
```diff
@@ -5,8 +5,8 @@ import { mapGetters } from 'vuex';
 import { getContrastingTextColor } from '@chatwoot/utils';
 import { isEmptyObject } from 'widget/helpers/utils';
 import { getRegexp } from 'shared/helpers/Validators';
-import { useMessageFormatter } from 'shared/composables/useMessageFormatter';
-import routerMixin from 'widget/mixins/routerMixin';
+import messageFormatterMixin from 'shared/mixins/messageFormatterMixin';
```
Comment (iamsivin) [inline app/javascript/widget/components/PreChat/Form.vue:8]: @fayazara The `messageFormatterMixin` is already removed, right?

**Review by iamsivin — COMMENTED** [review]
@fayazara LGTM, added a comment, please take a look.

File: app/javascript/widget/components/PreChat/Form.vue (hunk 2 above)
Comment (fayazara) [reply app/javascript/widget/components/PreChat/Form.vue]: @iamsivin fixed this, PTAL

**Review by iamsivin — APPROVED** [review]
LGTM

**Pull request closed without merging**